packages/aws-cdk-lib/aws-stepfunctions/lib/states/map-base.ts

```diff
@@ -0,0 +1,195 @@
+   *
+   * @default No comment
+   */
+  readonly comment?: string;
+
+  /**
+   * JSONPath expression to select part of the state to be the input to this state.
+   *
+   * May also be the special value JsonPath.DISCARD, which will cause the effective
+   * input to be the empty object {}.
+   *
+   * @default $
+   */
+  readonly inputPath?: string;
+
+  /**
+   * JSONPath expression to select part of the state to be the output to this state.
+   *
+   * May also be the special value JsonPath.DISCARD, which will cause the effective
+   * output to be the empty object {}.
+   *
+   * @default $
+   */
+  readonly outputPath?: string;
+
+  /**
+   * JSONPath expression to indicate where to inject the state's output
+   *
+   * May also be the special value JsonPath.DISCARD, which will cause the state's
+   * input to become its output.
+   *
+   * @default $
+   */
+  readonly resultPath?: string;
+
+  /**
+   * JSONPath expression to select the array to iterate over
+   *
+   * @default $
+   */
+  readonly itemsPath?: string;
+
+  /**
+   * The JSON that you want to override your default iteration input (mutually exclusive  with `parameters`).
+   *
+   * @see
+   * https://docs.aws.amazon.com/step-functions/latest/dg/input-output-itemselector.html
+   *
+   * @default $
+   */
+  readonly itemSelector?: { [key: string]: any };
+
+  /**
+   * The JSON that will replace the state's raw result and become the effective
+   * result before ResultPath is applied.
+   *
+   * You can use ResultSelector to create a payload with values that are static
+   * or selected from the state's raw result.
+   *
+   * @see
+   * https://docs.aws.amazon.com/step-functions/latest/dg/input-output-inputpath-params.html#input-output-resultselector
+   *
+   * @default - None
+   */
+  readonly resultSelector?: { [key: string]: any };
+
+  /**
+   * MaxConcurrency
+   *
+   * An upper bound on the number of iterations you want running at once.
+   *
+   * @default - full concurrency
+   */
+  readonly maxConcurrency?: number;
+}
+
+/**
+ * Returns true if the value passed is a positive integer
+ * @param value the value to validate
+ */
+export const isPositiveInteger = (value: number) => {
+  const isFloat = Math.floor(value) !== value;
+
+  const isNotPositiveInteger = value < 0 || value > Number.MAX_SAFE_INTEGER;
+
+  return !isFloat && !isNotPositiveInteger;
+};
+
+/**
+ * Define a Map state in the state machine
+ *
+ * A `Map` state can be used to run a set of steps for each element of an input array.
+ * A Map state will execute the same steps for multiple entries of an array in the state input.
+ *
+ * While the Parallel state executes multiple branches of steps using the same input, a Map state
+ * will execute the same steps for multiple entries of an array in the state input.
+ *
+ * @see https://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-map-state.html
+ */
+export abstract class MapBase extends State implements INextable {
```

Comment:
Whats the motivation behind moving `MapBase` out of `Map`?

Reply:
With this PR, I wanted to do two things:
- Implement a new Distributed Maps construct
- Ensure that this new construct does not support deprecated ASL fields.

The way I decided to go about this was to abstract the common properties of both distributed and inline maps, and only support deprecated ASL fields (`parameters` and `iterator`) in the existing Map construct for backwards compatibility concerns.

Two other alternatives that I briefly considered but ultimately rejected:
- Adding to the existing Map construct the required Distributed Map fields. This was rejected as having fields be conditionally available based on what kind of map it is seemed unnecessarily convoluted
- Extending the existing Map construct to create the Distributed Map one. This was rejected as the new construct would contain deprecated ASL fields